src/components/models/CartModel.ts

```diff
@@ -0,0 +1,232 @@
+        } else {
+            this.totalSum = 0;
+        }
+        return this.totalSum;
+    }
+
+    private checkTotalSumWithDiscountInLocalStorage() {
+        const totalSumWithDiscountInLocalStorage = window.localStorage.getItem('totalSumWithDiscount');
+        if (totalSumWithDiscountInLocalStorage) {
+            this.totalSumWithDiscount = Number(totalSumWithDiscountInLocalStorage);
+        } else {
+            this.totalSumWithDiscount = this.totalSum;
+        }
+        return this.totalSumWithDiscount;
+    }
+
+    private checkProductQuantityInLocalStorage() {
+        const productQuantityInLocalStorage = localStorage.getItem('productsQuantity');
+        if (productQuantityInLocalStorage) {
+            this.productsQuantity = Number(productQuantityInLocalStorage);
+        } else {
+            this.productsQuantity = 0;
+        }
+        return this.productsQuantity;
+    }
+
+    addProduct(product: Product) {
+        const isInCart = this.checkProductInCart(product.id);
+        if (!isInCart) {
+            this.productsInCart.push(product);
+            product.inCart = 1;
+            product.sum = product.price;
+            this.productsQuantity += product.inCart;
+            this.totalSum = Number((this.totalSum += product.inCart * product.price).toFixed(2));
+            this.appController.addProductToCart(product);
+            this.calculateAddingSumDiscoutDelete();
+            this.appController.cartView.updateDiscountInfo();
+            return this.productsInCart;
+        }
+    }
+
+    deleteProduct(product: Product) {
+        const isInCart = this.checkProductInCart(product.id);
+        if (isInCart) {
+            const deletedIndex = this.productsInCart.indexOf(product);
+            this.productsInCart.splice(deletedIndex, 1);
+            if (product.inCart) {
+                this.productsQuantity -= product.inCart;
+                this.totalSum = Number((this.totalSum -= product.inCart * product.price).toFixed(2));
+                product.inCart = 0;
+                this.appController.cartView.totalPerProduct = 0;
+                const currentTotalPercentDiscount = this.getCurrentTotalPercentDiscount();
+                this.totalSumWithDiscount = Number(
+                    (this.totalSum * (1 - currentTotalPercentDiscount / 100)).toFixed(2)
+                );
+                this.appController.cartView.updateDiscountInfo();
+            }
+        }
+        return this.productsInCart;
+    }
+
+    public addOneProduct(product: Product) {
+        if (product.inCart && product.sum && product.inCart < product.quantity) {
+            product.inCart += 1;
+            this.productsQuantity += 1;
+            this.totalSum = Number((this.totalSum += product.price).toFixed(2));
+            product.sum = Number((product.sum += product.price).toFixed(2));
+            this.calculateAddingSumDiscoutDelete();
+            this.appController.cartView.updateCartInfo();
```

Comment:
I suggest to put these lines : 
           this.appController.cartView.updateCartInfo();
            this.appController.cartView.updatePromoBlock();
            this.appController.cartView.updatePage();
            this.appController.cartView.updateDiscountInfo();
into separate method and reuse it.

Reply:
Fix it